### PR TITLE
Fix names and update documentation for the demo retention curve model

### PR DESCRIPTION
**Pull request by dse0**
Some of the attribute and metric names didn't seem right to me, please check that I fixed those correctly. I also have some questions about this model... when are `cohort_month_str` and `cohort_year_str` necessary? Can't we just use `cohort_month` and limit it to Month, Quarter, or Year granularity?

### CODE REVIEW

File: 02_retention_curves/models/user_month_retention.yml [hunk 1]
```diff
@@ -40,42 +41,53 @@ cols:
...
   - id: signup_year
     type: attribute
     physicalName: cohort_year_str
-    name: signup year
+    name: Signup year
+    description: String version of the signup year
   - id: row_count
     type: metric
-    name: __analytic rows
+    name: Month count
```
Comment (vndrewlee) [inline 02_retention_curves/models/user_month_retention.yml:74]: each row in the underlying table is (month, user_id) resolution, so month count is not accurate

**Comment by dse0** [comment]
This is ready for another look. I got rid of the `signup_month` and `signup_year` attributes after confirming with @trucklos . Preview: https://demo.glean.io/app/p/builds/_8FHwijeVYcvgbVe

File: 02_retention_curves/models/user_month_retention.yml [hunk 2]
```diff
@@ -50,40 +52,41 @@ cols:
     physicalName: cohort_month
     name: Cohort
```
Comment (vndrewlee) [inline 02_retention_curves/models/user_month_retention.yml:53]: suggestion, rename to "signup cohort"

File: 02_retention_curves/saved_views/cohort_table.yml [hunk 3]
```diff
@@ -1,41 +1,38 @@
-glean: '1.0'
+glean: "1.0"
```
Comment (vndrewlee) [inline 02_retention_curves/saved_views/cohort_table.yml:1]: Screenshot 2023-05-02 at 6 27 12 PM

I see this error in the preview, expected?

**Review by vndrewlee — APPROVED** [review]

File: 02_retention_curves/saved_views/cohort_table.yml (hunk 3 above)
Comment (dse0) [reply 02_retention_curves/saved_views/cohort_table.yml:1]: Appears to be the ongoing bigquery / google issue.

**Pull request merged**